build_tools/github_actions/lint_whitespace_checks.sh

```diff
@@ -40,7 +40,7 @@ if [[ $# -ne 0 ]] ; then
 fi
 
 echo "Gathering changed files..."
-mapfile -t CHANGED_FILES < <(git diff "$BASE_BRANCH" HEAD --name-only --diff-filter=d | grep '.*\.cpp$\|.*\.h$\|.*\.md$\|.*\.mlir$\|.*\.sh$\|.*\.td$\|.*\.txt$\|.*\.yml$\|.*\.yaml$')
+mapfile -t CHANGED_FILES < <(git diff "$BASE_BRANCH" HEAD --name-only --diff-filter=d | grep -E '.*\.(cpp|h|md|mlir|sh|td|txt|ya?ml|py)$')
```

Comment:
In our repo, since every file (except `.bc`) should have terminating newline, wdyt about changing this to exclude list? This would also simplify the grep script to something like:
`grep -v '\.bc$'`

Reply:
This is a great point! Here is the list of extensions we use:

```
.bazel
.bc
.bzl
.cmake
.cpp
.h
.in
.json
.lock
.md
.mlir
.pdll
.png
.py
.sh
.svg
.td
.txt
.yaml
```

Of those, I think `.bc`, `.in` (generated files), `.png` and `.svg` can be safely excluded.

Indeed, way more files and file extensions should be checked for whitespace than not. In the future, if new file extensions come up, we will likely want them to be checked rather than excluded, so using an exclude list makes sense. I will make the change.

Reply:
We could do allowlist or denylist. Maybe we try the denylist for a bit. Currently in our repo, filetypes that don't need to end in newline are: png, svg, bc.

If someone uploaded a jpeg during RFC process, this lint will fail. Maybe that's OK at the cost of a more sound lint?

Reply:
Didn't notice the new comment, on the same page :)